### PR TITLE
Chat SDK uses Communication identifiers

### PR DESCRIPTION
**Pull request by knvsl**
Chat convenience layer should return id's as CommunicationUserIdentifier's instead of strings.

- Introduced custom models that use the communication identifier
- Convenience layer calls the generated code and receives a response in the format of one of the generated models, it now maps the generated model to the custom model that uses identifiers and returns that instead
- For PagedCollection have gone with the workaround of throwing away the paged collection that the generated code returns and constructing a new one from the data within the httpResponse: https://github.com/Azure/azure-sdk-for-ios/issues/644

### CODE REVIEW

File: sdk/communication/AzureCommunicationChat/Source/ChatThreadClient.swift [hunk 1]
```diff
@@ -276,30 +330,53 @@ public class ChatThreadClient {
...
     public func listMessages(
         withOptions options: ChatThreadOperation.ListChatMessagesOptions? = nil,
-        completionHandler: @escaping HTTPResultHandler<PagedCollection<ChatMessage>>
+        completionHandler: @escaping HTTPResultHandler<PagedCollection<Message>>
     ) {
         service.listChatMessages(chatThreadId: threadId, withOptions: options) { result, httpResponse in
             switch result {
-            case let .success(messages):
-                completionHandler(.success(messages), httpResponse)
+            case .success: https: // github.com/Azure/azure-sdk-for-ios/issues/644
```
Comment (sacheu) [inline sdk/communication/AzureCommunicationChat/Source/ChatThreadClient.swift:337]: https: should be in a comment.

File: sdk/communication/AzureCommunication/Source/Identifiers.swift [hunk 2]
```diff
@@ -33,7 +33,7 @@ import Foundation
 /**
  Communication identifier for Communication Services Users
  */
-@objcMembers public class CommunicationUserIdentifier: NSObject, CommunicationIdentifier {
+@objcMembers public class CommunicationUserIdentifier: NSObject, Codable, CommunicationIdentifier {
```
Comment (JoshuaLai) [inline sdk/communication/AzureCommunication/Source/Identifiers.swift:36]: @sacheu this introduction of `Codable` careful need to take into consideration for https://github.com/Azure/azure-sdk-for-ios/pull/655

File: sdk/communication/AzureCommunicationChat/Source/ChatThreadClient.swift (hunk 1 above)
Comment (knvsl) [reply sdk/communication/AzureCommunicationChat/Source/ChatThreadClient.swift]: oh weird, fixed. thanks!

File: sdk/communication/AzureCommunication/Source/Identifiers.swift (hunk 2 above)
Comment (knvsl) [reply sdk/communication/AzureCommunication/Source/Identifiers.swift:36]: I think I can actually revert this, when encoding/decoding the custom models not actually encoding/decoding CommunicationUserIdentifier

**Review by glorialimicrosoft — APPROVED** [review]

**Pull request merged**